### ISSUE TITLE
Local exception error when using ppx_sexp_conv in a Docker container

### ISSUE THREAD

**Issue by antoine1fr** [janestreet/ppx_sexp_conv#17]
When bulding a project in a container using ocaml >= 4.04, oasis and ppx_sexp_conv, I get the following strange error:

```
# make
ocaml setup.ml -build 
Finished, 0 targets (0 cached) in 00:00:00.
+ /root/.opam/4.04.2/bin/ocamlfind ocamldep -package ppx_sexp_conv -modules src/test.ml > src/test.ml.depends
File "src/test.ml", line 3, characters 2-25: local exceptions are not supported before OCaml 4.04
File "src/test.ml", line 1:
Error: Error while running external preprocessor
Command line: /root/.opam/4.04.2/lib/ppx_deriving/./ppx_deriving package:ppx_sexp_conv '/tmp/camlppx711eff' '/tmp/camlppx9cd83f'

Command exited with code 2.
Compilation unsuccessful after building 1 target (0 cached) in 00:00:00.
E: Failure("Command ''/root/.opam/4.04.2/bin/ocamlbuild' src/test.cma src/test.cmxa src/test.a src/test.cmxs -use-ocamlfind -tag debug' terminated with error code 10")
Makefile:7: recipe for target 'build' failed
make: *** [build] Error 1
```

In src/test.ml:
```
let () =
  let exception E in
  ()
```

In _oasis:
```
OASISFormat: 0.4
Name:        test
Version:     0.1.0
Synopsis:    This is a test.
Authors:     Antoine
License:     MIT
Plugins:     META (0.4), DevFiles (0.4)
BuildTools: ocamlbuild
OCamlVersion: >= 4.04

Library test
  Path: src
  Modules: Test
  BuildDepends:
    ppx_sexp_conv
```

The problem disappears when dropping the ppx_sexp_conv dependency.

I can reproduce this behavior in containers using alpine:latest and ubuntu:latest (xenial), but not on my real development environment (an Arch Linux).

**Comment by antoine1fr** [comment]
Note there is no need for oasis to reproduce. The problem also arises with jbuilder and ocamlbuild.

**Comment by antoine1fr** [comment]
Ok it seems to be due to ppx_deriving 4.2 that has been released 2 days ago. When switching back to 4.1 it works again.

**Comment by antoine1fr** [comment]
Today I reproduced the error with a little project, using jbuilder, ppx_sexp_conv and ocaml 4.04.2.

Here is my jbuild file:
```lisp
(jbuild_version 1)

(executable
  ((name test)
   (libraries ())
   (preprocess (pps (ppx_sexp_conv)))))
```
And here is my OCaml file:
```ocaml
let () =
  let exception E of int in
  ()
```
I get the following error log:
```
(cd _build/default && ./.ppx/ppx_sexp_conv/ppx.exe --dump-ast -o lib/test.pp.ml --impl lib/test.ml)
Fatal error: exception File "lib/test.ml", line 8, characters 2-32: local exceptions are not supported before OCaml 4.04
Raised at file "src/migrate_parsetree_404_403_migrate.ml", line 24, characters 2-49
Called from file "src/migrate_parsetree_404_403_migrate.ml", line 37, characters 8-49
Called from file "src/migrate_parsetree_404_403_migrate.ml", line 244, characters 8-34
Called from file "list.ml", line 59, characters 20-23
Called from file "src/migrate_parsetree_404_403_migrate.ml", line 502, characters 10-42
Called from file "src/migrate_parsetree_404_403_migrate.ml", line 485, characters 8-44
Called from file "list.ml", line 59, characters 20-23
Called from file "src/migrate_parsetree_driver.ml", line 216, characters 21-67
Called from file "src/migrate_parsetree_driver.ml", line 222, characters 11-86
Called from file "src/migrate_parsetree_driver.ml", line 370, characters 10-60
Called from file "list.ml", line 77, characters 12-15
Called from file "src/migrate_parsetree_driver.ml", line 445, characters 4-86
Re-raised at file "parsing/location.ml", line 450, characters 12-23
Re-raised at file "parsing/location.ml", line 450, characters 12-23
Re-raised at file "parsing/location.ml", line 450, characters 12-23
Re-raised at file "parsing/location.ml", line 450, characters 12-23
Re-raised at file "parsing/location.ml", line 450, characters 12-23
Re-raised at file "parsing/location.ml", line 450, characters 12-23
Called from file "parsing/location.ml" (inlined), line 454, characters 31-61
Called from file "src/migrate_parsetree_driver.ml", line 447, characters 4-54
Called from file "src/migrate_parsetree_driver.ml", line 475, characters 4-31
Called from file "src/migrate_parsetree_driver_main.ml", line 1, characters 9-45
```

**Comment by antoine1fr** [comment]
I'm using ppx_sexp_conv v0.9.0.

**Comment by prepor** [comment]
I have the same problem

**Comment by hhugo** [comment]
ppx packages with version v0.9 were based on the ast of ocaml 4.03 (where local exceptions don't exists).
It should be fixed with the latest release (v0.10)
